snapcraft/models/project.py

```diff
@@ -1261,6 +1290,9 @@ def get_build_plan(self) -> List[BuildInfo]:
         # set default value
         if self.platforms is None:
             self.platforms = {get_host_architecture(): None}
+            # For backwards compatibility with core22, convert the platforms.
+            if self.base == "core22" and self.architectures:
```

Comment:
> self.base

Shouldn't we use the "effective" base here? For instance, this is a project that should fall into this right?

```yaml
base: bare
build-base: core22

Reply:
Good point! I actually just repurposed `effective_base` from a few lines above since that just gets us a string of "22.04" and does exactly what we need otherwise.